src/core/util/filesystem.cc

```diff
@@ -96,21 +96,20 @@ StringStream::ReadKeyPair()
 
   // TODO(anonimal): debug logging; include delimiter/terminator
 
-  return std::make_tuple(key, value, read_size);
+  return std::make_tuple(std::move(key), std::move(value), read_size);
 }
 
-const std::string StringStream::ReadStringFromByte()
+std::string StringStream::ReadStringFromByte()
 {
   // Get stated length amount
   std::uint8_t len;
   m_Stream.read(reinterpret_cast<char*>(&len), 1);
 
+  std::string string(static_cast<std::string::size_type>(len), '\0');
```

Comment:
>static_cast<<std::string::size_type>>()

This isn't necessary with gcc or clang. Is this another stupid MSVC caveat? If not, please don't give me a pedantic lecture about something that isn't necessary.

Reply:
You're right, this code isn't necessary.